realm/src/main/java/io/realm/RealmResults.java

```diff
@@ -70,7 +71,23 @@
     private Future<Long> pendingQuery;
     private boolean isCompleted = false;
 
-    RealmResults(Realm realm, Class<E> classSpec) {
+    static <E extends RealmObject> RealmResults<E> createFromClass(BaseRealm realm, Class<E> clazz) {
+        return new RealmResults<E>(realm, clazz);
+    }
+
+    static <E extends RealmObject> RealmResults<E> createFromQuery(BaseRealm realm, TableOrView table, Class<E> clazz) {
```

Comment:
This method doesn't use a query to create a `RealmResults`. A better name might be createFromTableOrView`.